haystack/modeling/model/optimization.py

```diff
@@ -310,24 +280,23 @@ def optimize_model(
     :param device: either torch.device("cpu") or torch.device("cuda"). Get the device from `initialize_device_settings()`
     :param distributed: Whether training on distributed machines
     :param local_rank: rank of the machine in a distributed setting
-    :param use_amp: Optimization level of nvidia's automatic mixed precision (AMP). The higher the level, the faster the model.
-                    Options:
-                    "O0" (Normal FP32 training)
-                    "O1" (Mixed Precision => Recommended)
-                    "O2" (Almost FP16)
-                    "O3" (Pure FP16).
-                    See details on: https://nvidia.github.io/apex/amp.html
+    :param optimizer: optimizer
+    :param use_amp: This option is deprecated. Only Pytorch automatic mixed precision (AMP) is supported. The Apex
+                    library is no longer supported. This means that `use_amp` is no longer used by this function
+                    since it is not needed to initialize native Pytorch AMP. If a value is provided a warning message
+                    will be raised.
     :return: model, optimizer
     """
-    model, optimizer = _init_amp(model, device, optimizer, use_amp)
+    if use_amp:
+        logger.warning(
```

Comment:
I have the same comment here as for the warning above. 🙂

Reply:
Do you think we need the warning both in `initialize_optimizer` and `optimize_model`?

Reply:
I think it's nice to have in the off chance that a user decides to call `initialize_optimizer` and `optimize_model` explicitly instead of just using the built-in `train` methods.

Without the warning message, I would recommend we remove `use_amp` here and list this as a breaking change.

Reply:
Alright, I think it's okay to show the warning twice.